invenio_rdm_records/collections/tasks.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Collections celery tasks."""
+
+from celery import shared_task
+from flask import current_app
+from invenio_access.permissions import system_identity
+
+from invenio_rdm_records.proxies import current_rdm_records
+
+
+@shared_task(ignore_result=True)
+def update_collections_size():
+    """Calculate and update the size of all the collections."""
+    collections_service = current_rdm_records.collections_service
+    res = collections_service.read_all(system_identity, depth=0)
+    for citem in res:
+        try:
+            collection = citem._collection
+            res = collections_service.search_records(system_identity, collection)
```

Comment:
does it make sense to take advantage of the OS count query instead?

Reply:
I am not sure whether it's faster or not (I assume it should be, but to be tested), but is it supported by the other services methods? Searching for records inside a collection ultimately means using the `CommunityRecords` or `RecordsService` search.